Vates/VatesAPI/src/vtkPeakMarkerFactory.cpp

```diff
@@ -108,12 +72,109 @@ namespace VATES
     validateWsNotNull();
   }
 
+  /**
+   Get ellipsoid axes
+   @param ellipticalShape: ellipsoidal parameters.
+   @param peak: peak under consideration.
+   */
+  std::vector<Mantid::Kernel::V3D> vtkPeakMarkerFactory::getAxes(
+      const Mantid::DataObjects::PeakShapeEllipsoid &ellipticalShape,
+      const IPeak &peak) const {
+    std::vector<Mantid::Kernel::V3D> axes;
+    switch (m_dimensionToShow) {
+    case vtkPeakMarkerFactory::Peak_in_Q_lab:
+      axes = ellipticalShape.directions();
+      break;
+    case vtkPeakMarkerFactory::Peak_in_Q_sample: {
+      Mantid::Kernel::Matrix<double> goniometerMatrix =
+          peak.getGoniometerMatrix();
+      if (goniometerMatrix.Invert() != 0.0) {
+        axes = ellipticalShape.getDirectionInSpecificFrame(goniometerMatrix);
+      } else {
+        axes = ellipticalShape.directions();
+      }
+    } break;
+    case vtkPeakMarkerFactory::Peak_in_HKL:
+      axes = ellipticalShape.directions();
+      break;
+    default:
+      axes = ellipticalShape.directions();
+    }
+    return axes;
+  }
+
+  /**
+   Get glyph position in the appropiate coordinates.
+   @param peak: peak under consideration.
+   */
+  V3D vtkPeakMarkerFactory::getPosition(const IPeak &peak) const {
+
+    V3D pos;
+    switch (m_dimensionToShow) {
+    case Peak_in_Q_lab:
+      pos = peak.getQLabFrame();
+      break;
+    case Peak_in_Q_sample:
+      pos = peak.getQSampleFrame();
+      break;
+    case Peak_in_HKL:
+      pos = peak.getHKL();
+      break;
+    default:
+      pos = peak.getQLabFrame();
+    }
+    return pos;
+  }
 
-    /**
-  Create the vtkStructuredGrid from the provided workspace
-  @param progressUpdating: Reporting object to pass progress information up the stack.
-  @return vtkPolyData glyph.
-  */
+  /**
+   Get the tranform tensor for vtkTensorGlyph
+   @param ellipticalShape: ellipsoidal parameters.
+   @param peak: peak under consideration.
+   */
+  std::array<float, 9> vtkPeakMarkerFactory::getTransformTensor(
+      const Mantid::DataObjects::PeakShapeEllipsoid &ellipticalShape,
+      const IPeak &peak) const {
+
+    std::vector<double> radii = ellipticalShape.abcRadii();
+    std::vector<Mantid::Kernel::V3D> axes = getAxes(ellipticalShape, peak);
+
+    std::array<float, 9> tensor;
+    for (unsigned j = 0; j < 3; ++j) {
+      for (unsigned k = 0; k < 3; ++k) {
+        tensor[3 * j + k] = static_cast<float>(radii[j] * axes[j][k]);
```

Comment:
Maybe also good to document what you are doing here. It might not be obvious that you are creating a (1) scaling by the radii and then (2) rotation into the eigenframe.